LiteDB.Benchmarks/Benchmarks/GeneralMethods/StopwatchExtensionsBenchmark.cs

```diff
@@ -1,43 +1,128 @@
 ﻿using BenchmarkDotNet.Attributes;
 using BenchmarkDotNet.Jobs;
+
 using LiteDB.Utils.Extensions;
+
 using System;
+using System.Collections;
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
 
-namespace LiteDB.Benchmarks.Benchmarks.GeneralMethods
+namespace LiteDB.Benchmarks.Benchmarks.GeneralMethods;
+
+[MemoryDiagnoser]
+[SimpleJob(RuntimeMoniker.Net472, baseline: true)]
+[SimpleJob(RuntimeMoniker.Net60)]
+[BenchmarkCategory(Constants.Categories.GENERAL)]
+public class StopwatchExtensionsBenchmark
 {
-	[MemoryDiagnoser]
-	[SimpleJob(RuntimeMoniker.Net472, baseline: true)]
-	[SimpleJob(RuntimeMoniker.Net60)]
-	public class StopwatchExtensionsBenchmark
-	{
-		Stopwatch stopwatch = new Stopwatch();
-		Action a = () => { };
-
-		[Benchmark]
-		public StopWatchExtensions.DisposableAction ReturningNewDisposableAction()
-		{
-			return stopwatch.StartDisposable();
-		}
-
-		[Benchmark]
-		public int Sum()
-		{
-			var watchDisposable = stopwatch.StartDisposable();
-			int sum = 0;
-			try
-			{
-				foreach (var i in Enumerable.Range(0, 100))
-				{
-					sum += i;
-				}
-			}
-			finally
-			{
-				watchDisposable.Dispose();
-			}
-			return sum;
-		}
-	}
-}
+    private Stopwatch stopwatch = new Stopwatch();
+    private Action a = () => { };
+
+    [Benchmark]
+    public StopWatchExtensions.DisposableAction ReturningNewDisposableAction()
+    {
+        return stopwatch.StartDisposable();
+    }
+
+    [Benchmark]
+    public int Sum()
+    {
+        var watchDisposable = stopwatch.StartDisposable();
+        int sum = 0;
+        try
+        {
+            foreach (var i in Enumerable.Range(0, 100))
+            {
+                sum += i;
+            }
+        }
+        finally
+        {
+            watchDisposable.Dispose();
+        }
+        return sum;
+    }
+
+    [Benchmark]
+    public int Struct_SumWithRangeHavingSw()
+    {
+        int sum = 0;
+        foreach (var i in Struct_Range(0, 100))
+        {
+            sum += i;
+        }
+        return sum;
+    }
+
+    private static IEnumerable<int> Struct_Range(int start, int count)
+    {
+        var sw = new Stopwatch();
+
+        using var _ = sw.StartDisposable();
```

Comment:
of course this will be slower, it's boxing the struct no matter what. So, you're ignoring the actual change. This will lowering to a cast into the `IDisposable` interface